Src/ILGPU/ILGPU.csproj

```diff
@@ -50,7 +50,6 @@
     </PackageReference>
     <PackageReference Include="System.Collections.Immutable" Version="7.0.0" />
     <PackageReference Include="System.Memory" Version="4.5.5" />
-    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="6.0.0" />
```

Comment:
Is this really not required for legacy-builds using older SDKs?

Reply:
It is a transitive dependency of ` System.Collections.Immutable` and `System.Memory`.

In net70, `System.Collections.Immutable` no longer depends on `System.Runtime.CompilerServices.Unsafe` (there is no package for net70).

QUESTION: Why do we depend on these "System" nuget packages? And not just use the version provided in the .NET framework?

Reply:
It looks like `System.Collections.Immutable` and `System.Reflection.Metadata` are required for `net471` and `netstandard2.1`. For the others (`net50`, `net60`, `net70`), we can omit the nuget packages and compile without issue.

`System.Memory` and `System.Runtime.CompilerServices.Unsafe` are transitive dependencies of `System.Collections.Immutable`, and can be excluded.